ts/private/ts_project.bzl

```diff
@@ -0,0 +1,177 @@
+"""ts_project implementation
+
+Nearly identical to the ts_project wrapper macro in rules_nodejs.
```

Comment:
Is there a 1-line summary of what is different from rules_nodejs that could be added here?

Reply:
yeah good suggestion